sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -488,12 +497,15 @@ def run_pipeline(self, pipeline, options):
       # Cross language transform require using a pipeline object constructed
       # from the full pipeline proto to make sure that expanded version of
       # external transforms are reflected in the Pipeline job graph.
+      # TODO(chamikara): remove following pipeline and pipeline proto recreation
+      # after portable job submission path is fully in place.
       from apache_beam import Pipeline
       pipeline = Pipeline.from_runner_api(
           self.proto_pipeline,
           pipeline.runner,
           options,
           allow_proto_holders=True)
+      self._adjust_types_for_dataflow(pipeline)
```

Comment:
Why do the type encodings not survive the round trip (pipeline -> proto -> pipeline)?

Reply:
Verified that types are preserved in the pipeline->proto->pipeline roundtrip and removed this.